api/buffer/design.go

```diff
@@ -0,0 +1,549 @@
+		HTTP(func() {
+			GET("")
+			Meta("openapi:tag:documentation")
+			Response(StatusOK)
+		})
+	})
+
+	Method("HealthCheck", func() {
+		NoSecurity()
+		Result(String, func() {
+			Example("OK")
+		})
+		HTTP(func() {
+			GET("//health-check")
+			Meta("openapi:generate", "false")
+			Response(StatusOK, func() {
+				ContentType("text/plain")
+			})
+		})
+	})
+
+	Files("/documentation/openapi.json", "openapi.json", func() {
+		Meta("openapi:summary", "Swagger 2.0 JSON Specification")
+		Meta("openapi:tag:documentation")
+	})
+	Files("/documentation/openapi.yaml", "openapi.yaml", func() {
+		Meta("openapi:summary", "Swagger 2.0 YAML Specification")
+		Meta("openapi:tag:documentation")
+	})
+	Files("/documentation/openapi3.json", "openapi3.json", func() {
+		Meta("openapi:summary", "OpenAPI 3.0 JSON Specification")
+		Meta("openapi:tag:documentation")
+	})
+	Files("/documentation/openapi3.yaml", "openapi3.yaml", func() {
+		Meta("openapi:summary", "OpenAPI 3.0 YAML Specification")
+		Meta("openapi:tag:documentation")
+	})
+	Files("/documentation/{*path}", "swagger-ui", func() {
+		Meta("openapi:generate", "false")
+		Meta("openapi:summary", "Swagger UI")
+		Meta("openapi:tag:documentation")
+	})
+
+	// Main endpoints ---------------------------------------------------------------------------------------------
+
+	Method("CreateReceiver", func() {
+		Meta("openapi:summary", "Create receiver")
+		Description("Create a new receiver for a given project")
+		Result(Receiver)
+		Payload(func() {
+			receiverId()
+			name("receiver")
+			Attribute("exports", ArrayOf(Export), "List of receiver exports. A receiver may have a maximum of 20 exports.")
+			Required("name", "exports")
+		})
+		HTTP(func() {
+			POST("/receivers")
+			Meta("openapi:tag:receiver")
+			Response(StatusOK)
+		})
+	})
+
+	Method("ListReceivers", func() {
+		Meta("openapi:summary", "List all receivers")
+		Description("List all receivers for a given project.")
+		Result(ArrayOf(Receiver))
+		HTTP(func() {
+			GET("/receivers")
+			Meta("openapi:tag:receiver")
+			Response(StatusOK)
+		})
+	})
+
+	Method("GetReceiver", func() {
+		Meta("openapi:summary", "Get receiver")
+		Description("Get the configuration of a receiver.")
+		Result(Receiver)
+		Payload(func() {
+			receiverId()
+			Required("receiverId")
+		})
+		HTTP(func() {
+			GET("/receivers/{receiverId}")
+			Meta("openapi:tag:receiver")
+			Response(StatusOK)
+			ReceiverNotFoundError()
+		})
+	})
+
+	Method("DeleteReceiver", func() {
+		Meta("openapi:summary", "Delete receiver")
+		Description("Delete a receiver.")
+		Result(Receiver)
+		Payload(func() {
+			receiverId()
+			Required("receiverId")
+		})
+		HTTP(func() {
+			DELETE("/receivers/{receiverId}")
+			Meta("openapi:tag:receiver")
+			Response(StatusOK)
+			ReceiverNotFoundError()
+		})
+	})
+
+	Method("RefreshReceiverTokens", func() {
+		Meta("openapi:summary", "Refresh receiver tokens")
+		Description("Refresh all tokens used by a receiver.")
```

Comment:
tu by som mozno pridal `Each export uses its own token scoped to the target bucket.`